redbot/core/settings_caches.py

```diff
@@ -122,157 +122,145 @@ async def set_ignored_guild(self, guild: discord.Guild, set_to: bool):
             if gid is not None:
-                ret = await self._config.guild_from_id(gid).blacklist()
-                if not ret:
-                    ret = []
+                ret = set(await self._config.guild_from_id(gid).blacklist())
             else:
-                ret = await self._config.blacklist()
+                ret = set(await self._config.blacklist())
 
             self._cached_blacklist[gid] = ret.copy()
 
         return ret
 
-    async def add_to_blacklist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
+    async def add_to_blacklist(self, guild: Optional[discord.Guild], role_or_user: Iterator[int]):
         gid: Optional[int] = guild.id if guild else None
         role_or_user = role_or_user or []
-        if not isinstance(role_or_user, list) or not all(
+        if not isinstance(role_or_user, (set, list)) or not all(
             isinstance(r_or_u, int) for r_or_u in role_or_user
         ):
-            raise TypeError("Blacklisted objects must be a list of ints")
+            raise TypeError("Blacklisted objects must be a list or set of ints")
         if gid is None:
             if gid not in self._cached_blacklist:
-                self._cached_blacklist[gid] = await self._config.blacklist()
+                self._cached_blacklist[gid] = set(await self._config.blacklist())
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist[gid]:
-                    self._cached_blacklist[gid].append(obj_id)
-                    async with self._config.blacklist() as curr_list:
-                        curr_list.append(obj_id)
+                self._cached_blacklist[gid].add(obj_id)
+            await self._config.blacklist.set(list(self._cached_blacklist[gid]))
         else:
             if gid not in self._cached_blacklist:
                 self._cached_blacklist[gid] = await self._config.guild_from_id(gid).blacklist()
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist[gid]:
-                    self._cached_blacklist[gid].append(obj_id)
-                    async with self._config.guild_from_id(gid).blacklist() as curr_list:
-                        curr_list.append(obj_id)
+                self._cached_blacklist[gid].add(obj_id)
+            await self._config.guild_from_id(gid).blacklist.set(list(self._cached_blacklist[gid]))
 
     async def clear_blacklist(self, guild: Optional[discord.Guild] = None):
         gid: Optional[int] = guild.id if guild else None
-        self._cached_blacklist[gid] = []
+        self._cached_blacklist[gid] = set()
         if gid is None:
             await self._config.blacklist.clear()
         else:
             await self._config.guild_from_id(gid).blacklist.clear()
 
-    async def remove_from_blacklist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
+    async def remove_from_blacklist(
+        self, guild: Optional[discord.Guild], role_or_user: Iterator[int]
```

Comment:
Wrong type hint (also, look at the other comment below)
```suggestion
        self, guild: Optional[discord.Guild], role_or_user: Iterable[int]
```